src/http/api/resources/pin.js

```diff
@@ -57,7 +57,7 @@ exports.ls = {
     try {
       result = await ipfs.pin.ls(path, { type })
     } catch (err) {
-      throw Boom.boomify(err, { message: 'Failed to list pins' })
+      throw Boom.boomify(err)
```

Comment:
Removed message prefix to match error from go-ipfs.
Interop test for this is added in https://github.com/ipfs/interface-js-ipfs-core/pull/375/files#diff-6a00eb1307f801bb37999815584589bdR199